### ISSUE TITLE
Suggestions for new shortcuts

### ISSUE THREAD

**Issue by feiss** [aframevr/aframe-inspector#298]
Following #154 , support for more shortcuts would be very interesting for desktop users. 

(Some of these actions are not yet included in the editor, I'll try to write a spec for each one in a separated issue)
- [x]  **`w,e,r`**
  Switch to Translate/Rotate/Scale modes (Maya and Max standard, **very convenient**)
- [x] **`z`** 
  View selected object (**f** would be a more convenient shortcut for this, but is currently used for 'fullscreen') (see #300)
- [ ] **`a`** 
  View all objects (see #300 )
- [ ] **`ctrl + LMB`** 
  Add object to selection (this could be an whole issue itself, support for multiple selection)
- [ ] **`ctrl + a`**
  Select All
- [x] **`d`**
  Duplicate/clone current selection
- [x] **`g`** 
  Toggle grid visibility
- [x] **`n`**
  Add new entity
- [x]  **`supr/backspace`** 
  Delete current selection
- [x] **`ctrl+c, ctrl+v`**
  copy/paste current selection
- [x] **`1,2,3,4`**
  Switch to Front/Left/Top/Perspective view
- [x] **`o`**
  Toggle local/global transform mode
- [ ] **`Num +, Num -`** 
  Increase/decrease transform helper (in discrete steps, between (min, max) values)
- [x] **`ctrl+alt+i`** 
  Switch Edit and VR modes. 
- [x] **`Esc`** for exiting edit mode (current behaviour)
- [x] **`h`** 
  Toggle help/shortcuts panel

Shortcut customization in the cheatsheet panel that #276 suggested would be nice, although I don't think it's high priority.

**Comment by fernandojsg** [comment]
Thanks! Regarding the `esc` I was thinking the same but @dmarcos pointed that maybe the VR-scene is capturing `esc` by itself as it's a very generic shortcut so we can't just hijack it :(

**Comment by ngokevin** [comment]
When the inspector is active, the scene should be paused and event listeners should be removed, so keypresses should not do anything.

**Comment by fernandojsg** [comment]
@ngokevin yes that's why `esc` will works while the editor is opened, but if we exit the inspector and go back to the scene we should remove that listener

**Comment by feiss** [comment]
Can't you just add the event listener after the vr-scene and its events are initialized? (maybe with a setTimeout()?)

**Comment by fernandojsg** [comment]
@feiss is not a problem about timing is more a problem on what happens if the vrscene has a listener for esc that does something, if we add a listener we'll screw it, that's why we tend to use rare shortcuts in that situation like `ctrl+alt+i`.

**Comment by fernandojsg** [comment]
**h** to show shortcut cheatsheet

**Comment by feiss** [comment]
what about the traditional F1? with additional help links and info

**Comment by fernandojsg** [comment]
F1 is reserved usually by the browser, and for example in mac keyboard you need to click `Fn + F1`  as F1 by default is assigned to lower screen brightness.
Just as completely personal opinion I like having all the shortcuts close to each others so I could have my hand relaxed on the keyboard and just move the fingers to press each shortcut without need to add any movement to my wrist

**Comment by feiss** [comment]
I'm sold! **h** then

**Comment by fernandojsg** [comment]
Maybe `backspace` for delete? As mac keyboard doesn't have `supr` by default, you need to press `cmd + backspace`

**Comment by fernandojsg** [comment]
@feiss **Num +, Num -** Do you mean for example translate or rotate step by step? In that case in which axis?

**Comment by feiss** [comment]
backspace/supr: why not both?
num +, num- : no, just visually scaling up/down the helper, without affecting the selection.

**Comment by fernandojsg** [comment]
> backspace/supr: why not both?

Yep!

> num +, num- : no, just visually scaling up/down the helper, without affecting the selection.

I don't get that, right now the helper has a constant size, it's not affected by the zoom of the perspective view do you want to make it bigger/smaller? is that useful?

**Comment by fernandojsg** [comment]
By the way can you update the first comment adding `task lists` so we could keep checking the ones we're finishing? (https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)

**Comment by fernandojsg** [comment]
`ctrl+d` in windows is add bookmark right? What about using just `d` without a modifier?

**Comment by Coder206** [comment]
@fernandojsg Yes, `ctrl+d` in Windows adds a bookmark.

**Comment by feiss** [comment]
allrighty, `d` without modifier :)

About `num+,num-`, yup, just change up/down that constant size. You can usually change it in other 3d packages, it's an usability thing. Just think that the transform helper is the most important and used tool of the editor. Anyway, it's not very important, just a small thing.

**Comment by fernandojsg** [comment]
Updated the list of shortcuts created based on: https://github.com/aframevr/aframe-inspector/pull/305

**Comment by vershwal** [comment]
@feiss Could you check the copy/paste selection shortcut?
https://github.com/aframevr/aframe-inspector/pull/426

**Comment by feiss** [comment]
@vershwal this is current layout in master. Everything works nice. Fernando even added a `cut` functionality. Will be available very very soon, thanks! 👍 

![image](https://cloud.githubusercontent.com/assets/359872/22858332/702170ba-f0ba-11e6-9cd9-f9879d3a7f6c.png)

**Comment by vershwal** [comment]
Yes I saw the cut functionality. It works great.

Also by check I meant ticking the copy/paste dialogue in the issue above (Marking it as resolved). Wrong choice of words I guess.

I think you have checked it.

Thanks
